### ISSUE TITLE
Scripting: the API for getting PropertyTypes

### ISSUE THREAD

**Issue by finscn** [mapeditor/tiled#3419]
the 1.9 supplies the feature about PropertyTypes.
Bug there is no API for get the PropertyTypes informations.

And I can't get the display name of a custom  Enum-Property.
The value of it is number.

So, I hope there could be some APIs for that work.

**Comment by max-carroll-sky** [comment]
Yes would love this feature too, for a work around I'm doing this for the moment

```
  const projectFile = new TextFile("C:\\Users\\{my name}\\tiled-project\\untitled.tiled-project", TextFile.ReadOnly)
      const pfText = projectFile.readAll()
      const project = JSON.parse(pfText)

     const {propertyTypes} = project
```

Then I am getting the values later on like this

```
  const genericEnum = o.resolvedProperty('aiType')
        let aiType = ''
        if (genericEnum) {
            aiType = propertyTypes.find(p=> p.id === genericEnum.typeId).values[genericEnum.value]
        }
```

**Comment by Ishadijcks** [comment]
I would love if editing PropertyTypes was added to the API. I'd like to create an extension which synchronizes Tiled PropertyTypes with TypeScript enums

**Comment by bjorn** [comment]
> And I can't get the display name of a custom Enum-Property.
> The value of it is number.

@finscn While it's not entirely trivial yet, now there is an API available for this, for example with the `biome` property being a value from a custom enum, you can get it name as follows:

```js
var biome = tiled.activeAsset.resolvedProperty('biome');
var biomeType = tiled.project.findTypeByName(biome.typeName);
var biomeName = biomeType.nameOf(biome);
```

> I would love if editing PropertyTypes was added to the API. I'd like to create an extension which synchronizes Tiled PropertyTypes with TypeScript enums

@Ishadijcks If you get this to work based on the new API I'd love to see your solution!

We'd in any case welcome any feedback on the APIs added in #3971 before getting to the Tiled 1.12 release.
